### PR TITLE
BIP-85: Derive Entropy (from existing seed)

### PR DESCRIPTION
**Pull request by peter-conalgo**
Leveraging proposed BIP from <https://github.com/ethankosakovsky>, this new feature allows you to easily derive a seed phrase for another wallet based on the seed in Coldcard. Supports 12-, 18-, 24-word phrases, HDSEED value for Bitcoin Core, Raw XPRV export, and also 32- or 64-byte raw hex values for other purposes.

Bonus features:
- press (1) to save result to MicroSD card, in simple text file
- press (2) to switch to the derived key temporarily (until next reset)

Background text shown when you enter the menu:

```
Create Entropy for Other Wallets

This feature derives "entropy" based mathematically from this wallet's seed value.
This will be displayed as a 12 or 24 word seed phrase,
or formatted in other ways to make it easy to import into
other wallet systems.

You will be able to recreate this value later, based
only the seed-phrase or backup of this Coldcard.

There is no way to reverse the process, should the other wallet system be compromised,
so the other wallet is effectively segregated from the Coldcard and yet
also backed-up from it.
```

TODO still:
- [x] HDSEED case is probably wrong
- [x] xprv, and hex cases are ahead of the standard
- [x] test code
- [x] reach a consensus on XPRV usage
- [ ] test on real Coldcard hardware

### CODE REVIEW

**Comment by peter-conalgo** [comment]
Use `make && ./simulator.py --deriv` to start the simulator on the new menu item, with the XPRV from the BIP installed.

**Comment by ethankosakovsky** [comment]
Looks good!

File: shared/drv_entro.py [hunk 1]
```diff
@@ -0,0 +1,198 @@
+# (c) Copyright 2020 by Coinkite Inc. This file is part of Coldcard <coldcardwallet.com>
+# and is covered by GPLv3 license found in COPYING.
+#
+# BIP-(tbd): Deterministic Entropy From BIP32 Keychains
+#
+# Using the system's BIP32 master key, safely derive seeds phrases/entropy for other systems.
+#
+# see <https://gist.github.com/ethandrv_entro/268c52f018b94bea29a6e809381c05d6>
```
Comment (ethankosakovsky) [inline shared/drv_entro.py:8]: link is 404

**Comment by peter-conalgo** [comment]
Blocked still on:
- [ ] at least 2k of more flash space
- [ ] ideally, a BIP number
- [ ] good long-lasting URL for top-of-file comment

**Comment by peter-conalgo** [comment]
Concern: this feature doesn't belong in the "Danger Zone". It's pretty safe to use.

**Comment by ethankosakovsky** [comment]
This has been allocated as BIP 85.

**Pull request merged**